src/pages/cart/Cart.jsx

```diff
@@ -0,0 +1,210 @@
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import StarRatings from "react-star-ratings";
+import { Navbar } from "../../components/components-index";
+import { useAppContext } from "../../context/AppContext";
+import { Link } from "react-router-dom";
+import "./Cart.css";
+
+export const Cart = () => {
+  const encodedToken = localStorage.getItem("token");
+  axios.defaults.headers.common["authorization"] = encodedToken;
+
+  const { appState, appDispatch } = useAppContext();
+  const [cartItemValue, setCartItemValue] = useState(0);
+
+  useEffect(() => {
+    (async () => {
+      setCartItemValue(appState.cartItems);
+      try {
+        const response = await axios.get("/api/user/cart");
+        response && appDispatch({ type: "CART", payload: response.data.cart });
+      } catch (error) {
+        if (error.response.status === 500) {
+          alert("you need to login");
+          appDispatch({ type: "LOGIN-MODAL", payload: true });
+        } else if (error.response.status === 409) {
+          alert("already in wishlist");
+        } else {
+          console.log(error);
+        }
+      }
+    })();
+  }, [appDispatch, appState.cartItems]);
+
+  // const removeFromCart = async (appState.cart.id) => {
+  //   try {
+  //     const res = await axios.delete(`/api/user/cart/${_id}`);
+  //     if (res.status === 200) {
+  //       appDispatch({ type: "REMOVE-FROM-CART", payload: res.data.cart });
+  //     }
+  //   } catch (e) {
+  //     console.log(e);
+  //   }
+  // };
+  const priceHandler = {
+    price:
+      appState.cart &&
+      appState.cart.reduce(
+        (acc, item) => acc + Number(item.price) + appState.totalPrice,
+        0
+      ),
+    discount:
+      appState.cart &&
+      appState.cart.reduce(
+        (acc, item) =>
+          acc +
+          (Number(item.price) - (50 / 100) * Number(item.price)) +
+          appState.totalPrice,
+        0
+      ),
+    delivery:
+      appState.cart &&
+      appState.cart.reduce((acc, item) => acc + Number(item.price), 0) +
```

Comment:
I think it would be easier if you destructure appState rather than prefixing it before every use

Reply:
How can i destructure it!. Not getting!!

Reply:
probably you can do this way  eg: const { statevariable:{your state propes}, dispatchvar }